app/util/fareUtils.js

```diff
@@ -1,57 +1,49 @@
-import uniq from 'lodash/uniq';
+import { uniqBy } from 'lodash';
 
-// returns null or non-empty array of ticket names
-export function mapFares(fares, config) {
-  if (!Array.isArray(fares) || !config.showTicketInformation) {
-    return null;
-  }
-
-  const [regularFare] = fares.filter(fare => fare.type === 'regular');
-  if (!regularFare) {
-    return null;
-  }
-
-  const { components } = regularFare;
-  if (!Array.isArray(components) || components.length === 0) {
+export const getFaresFromLegs = (legs, config) => {
+  if (
+    !Array.isArray(legs) ||
+    legs.size === 0 ||
+    !config.showTicketInformation
+  ) {
     return null;
   }
+  const availableTickets = Object.values(config.availableTickets)
+    .map(r => Object.keys(r))
+    .flat();
+  const filteredLegs = legs.map(leg => ({
+    ...leg,
+    fareProducts: leg.fareProducts.filter(fp =>
+      availableTickets.includes(fp.product.id),
+    ),
+  }));
 
-  return components.map(fare => ({
-    ...fare,
-    agency:
-      (Array.isArray(fare.routes) &&
-        fare.routes.length > 0 &&
-        fare.routes[0].agency) ||
-      undefined,
+  const knownFareLegs = uniqBy(
+    filteredLegs
+      .filter(l => l.fareProducts.length > 0)
+      .map(leg => ({
```

Comment:
isn't fareProduct filtering  already done above?

Reply:
The first filtering only removes unavailable ticket types (for some reason OTP seems to return a fareProduct with id HSL:regular as well as the actual ticket for a leg with available tickets). The two subsequent filterings then separately process the legs that have available tickets and those that do not.